pkg/vmtest/integration.go

```diff
@@ -55,65 +53,45 @@ func main() {
 
 // Options are integration test options.
 type Options struct {
-	// Env is the Go environment to use to build u-root.
-	Env *golang.Environ
+	// BuildOpts are u-root initramfs options.
+	//
+	// Fields that are not set are populated by QEMU and QEMUTest as
+	// possible.
+	BuildOpts uroot.Opts
+
+	// QEMUOpts are QEMU VM options for the test.
+	//
+	// Fields that are not set are populated by QEMU and QEMUTest as
+	// possible.
+	QEMUOpts qemu.Options
+
+	// DontSetEnv doesn't set the BuildOpts.Env and uses the user-supplied one.
+	//
+	// TODO: make uroot.Opts.Env a pointer?
+	DontSetEnv bool
 
 	// Name is the test's name.
 	//
 	// If name is left empty, the calling function's function name will be
-	// used as determined by runtime.Caller
+	// used as determined by runtime.Caller.
 	Name string
 
-	// Go commands to include in the initramfs for the VM.
-	//
-	// If left empty, all u-root commands will be included.
-	Cmds []string
-
 	// Uinit are commands to execute after init.
 	//
-	// If populated, a uinit.go will be generated from these.
+	// If populated, a uinit.go will be generated from these and added to
+	// the busybox generated in BuildOpts.Commands.
 	Uinit []string
 
-	// Files are files to include in the VMs initramfs.
-	Files []string
-
-	// TmpDir is a temporary directory for build artifacts.
-	TmpDir string
-
-	// LogFile is a file to log serial output to.
-	//
-	// The default is serial/$Name.log
-	LogFile string
-
 	// Logger logs build statements.
 	Logger logger.Logger
 
-	// Timeout is the timeout for expect statements.
-	Timeout time.Duration
-
-	// Network is the VM's network.
-	Network *qemu.Network
-
 	// Extra environment variables to set when building (used by u-bmc)
 	ExtraBuildEnv []string
```

Comment:
`golang.Environ` works for me, but this does not - currently I use it to set build tags, rather than env vars.

Reply:
use what to set build tags? the comment here doesn't make sense in context of ExtraBuildEnv -- which I also want to remove, since u-bmc should have its own options

Reply:
As discussed in chat, what I want _is_ still possible. :+1: